### ISSUE TITLE
feat: YouTube Music: Allow playing "Content is unavailable" songs

### ISSUE THREAD

**Issue by csd-was-taken** [ReVanced/revanced-patches#2916]
### Feature description

In YouTube music, some tracks are greyed out and refuse to play, giving the error "Content is unavailable". I think this happens when the uploader doesn't properly flag it as music, but I'm not entirely sure. If there was some way to bypass this, that would be amazing.

Examples:
Most of the songs in this playlist: https://www.youtube.com/playlist?list=PLU9ps46KOfxR7mJi_l8DmwH8UJACGPFyk
https://www.youtube.com/watch?v=n5n7CSGPzqw
https://youtu.be/WqNjXP0qQVc (As far as i can tell this song only exists for this trailer)

### Motivation

I know I have weird music taste, but a lot of my music just won't play in YouTube music, so I'm forced to use either the main app, or download MP3s. Unblocking these songs that refuse to play for seemingly no reason would be a huge QOL feature.

### Acknowledgements

- [X] This issue is not a duplicate of an existing feature request.
- [X] I have chosen an appropriate title.
- [X] All requested information has been provided properly.

**Comment by oSumAtrIX** [comment]
I think the contents are blocked by region, ur forced to use a VPN.

**Comment by csd-was-taken** [comment]
no, that's not it. I can watch them fine on the main youtube app and website. and it would be weird for such arbitrary videos to be blocked in the US, and only on the music app

I set my VPN to The UK and Ukraine, and had the same results

**Comment by oSumAtrIX** [comment]
The Android client is different. This is blocked on YouTube's end. Figure out the reason then we can think about how much it makes sense to look into.
